src/cards_management.c

```diff
@@ -408,6 +408,80 @@ Card_t draw_one_card(void)
     return draw_deck->card;
 }
 
+/*----shuffle cards---------*/   
+/**
+ * @brief Recursively free the given deck
+ * @param d deck to be freed
+ */
+ 
+
+void free_deck(Deck_t* d) {
+    // Recursively free the deck
+    if (d != NULL) {
+        free_deck(d->next);
+        free(d);
+    }
+}
+
+
+/**
+ * @brief Swapping two Cards
+ * @param *a address of the first card.
+ * @param *b address of the sencond card.
+ */
+
+void swap(Card_t *a, Card_t *b) {
+    Card_t temp = *a;
+    *a = *b;
+    *b = temp;
+}
+
```

Comment:
We already have void swap_cards(Card_t *p_a, Card_t *p_b), please remove this redundant function, and call swap_cards.

Reply:
Duly noted